### ISSUE TITLE
evaluation not finishing...help

### ISSUE THREAD

**Issue by ashley915** [tensorflow/models#3927]
I'm having trouble evaluating my trained oxford pet data. 
I put my command in terminal like this:
python3 object_detection/eval.py \
    --logtostderr \
    --pipeline_config_path=/home/cjonrnd/models/research/object_detection/models/model/train_again/pipeline.config \
    --checkpoint_dir=/home/cjonrnd/models/research/object_detection/models/model/train_again/checkpoint \
    --eval_dir=/home/cjonrnd/models/research/object_detection/models/model/eval_again

and this runs with no problem, but stops after printing: See tf.nn.softmax_cross_entropy_with_logits_v2.
the terminal just sits there for like half an hour and stops running without any error message. 

I'm attaching a screenshot of my error. I'm using Ubuntu 16.04.3, Python 3.5, Tensorflow_gpu-1.5, and I am running all this locally. Please help me out.
Thank you!

![screenshot from 2018-04-10 15-26-56](https://user-images.githubusercontent.com/35290370/38539566-b42e8ba6-3cd3-11e8-8911-d3fc76fbac25.png)

**Comment by karmel** [comment]
It looks like the evaluation might be successfully finishing, just not printing to the command line. Try adding a call to print() around the evaluate results in your copy of [eval.py](https://github.com/tensorflow/models/blob/master/research/object_detection/eval.py#L130). Does that work?

**Comment by ashley915** [comment]
I already tried doing them and prints out fine. Just not getting the results. 
Am I supposed to get image files in the directory? What does the result look like?
![screenshot from 2018-04-11 09-36-33](https://user-images.githubusercontent.com/35290370/38590424-0e4acfd0-3d6c-11e8-9ef1-bb28d3423cd9.png)

**Comment by karmel** [comment]
Sorry, to clarify-- evaluator.evaluate returns metrics, and you should try to print those out to see if it's what you want.

In the last lines there, try:

```
metrics = evaluator.evaluate(create_input_dict_fn, model_fn, eval_config, categories,
                     FLAGS.checkpoint_dir, FLAGS.eval_dir)
print(metrics)
```

**Comment by ashley915** [comment]
It prints: None
I also tried printing individually:
  print(create_input_dict_fn)
  print(model_fn)
  print(eval_config)
  print(categories)
  print(FLAGS.checkpoint_dir)
  print(FLAGS.eval_dir)
but that all prints to the right directory.
![input_fun](https://user-images.githubusercontent.com/35290370/38595717-331c5a1a-3d88-11e8-9433-caec20fcdc56.png)
![model_fn1](https://user-images.githubusercontent.com/35290370/38595721-36ee2448-3d88-11e8-8b11-2a7b8a19dd33.png)
![model_fn2](https://user-images.githubusercontent.com/35290370/38595725-392df558-3d88-11e8-8b29-67bd32d8d5dd.png)
![eval_config](https://user-images.githubusercontent.com/35290370/38595729-3c16c20e-3d88-11e8-9cc8-9eaf58d57793.png)
![other](https://user-images.githubusercontent.com/35290370/38595732-3e9fc692-3d88-11e8-8cf5-b5ec8d18435b.png)

**Comment by karmel** [comment]
It's hard to debug without the code. A few questions:

1. Did you successfully train the model?
2. What images are you trying to evaluate?
3. Can you provide the text (not screenshots, please) of the relevant config files?
4. Have you tried running tensorboard?
5. Have you successfully run through and understood the provided [Jupyter notebook](https://github.com/tensorflow/models/blob/master/research/object_detection/object_detection_tutorial.ipynb)

**Comment by ashley915** [comment]
1. Yes. Training was finished successfully
2. I'm trying the oxford pet data (tf record file)
3. Here is the pipeline config file:

# Faster R-CNN with Resnet-152 (v1), configured for Oxford-IIIT Pets Dataset.
# Users should configure the fine_tune_checkpoint field in the train config as
# well as the label_map_path and input_path fields in the train_input_reader and
# eval_input_reader. Search for "PATH_TO_BE_CONFIGURED" to find the fields that
# should be configured.

model {
  faster_rcnn {
    num_classes: 37
    image_resizer {
      keep_aspect_ratio_resizer {
        min_dimension: 600
        max_dimension: 1024
      }
    }
    feature_extractor {
      type: 'faster_rcnn_resnet152'
      first_stage_features_stride: 16
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 300
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}

train_config: {
  batch_size: 1
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        manual_step_learning_rate {
          initial_learning_rate: 0.0003
          schedule {
            step: 900000
            learning_rate: .00003
          }
          schedule {
            step: 1200000
            learning_rate: .000003
          }
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  
  from_detection_checkpoint: false
  # Note: The below line limits the training process to 200K steps, which we
  # empirically found to be sufficient enough to train the pets dataset. This
  # effectively bypasses the learning rate schedule (the learning rate will
  # never decay). Remove the below line to train indefinitely.
  num_steps: 200000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
}

train_input_reader: {
  tf_record_input_reader {
    input_path: "/home/cjonrnd/models/research/object_detection/data/pet_train_with_masks.record"
  }
  label_map_path: "/home/cjonrnd/models/research/object_detection/data/pet_label_map.pbtxt"
}

eval_config: {
  num_examples: 2000
  # Note: The below line limits the evaluation process to 10 evaluations.
  # Remove the below line to evaluate indefinitely.
  max_evals: 10
}

eval_input_reader: {
  tf_record_input_reader {
    input_path: "/home/cjonrnd/models/research/object_detection/data/pet_val_with_masks.record"
  }
  label_map_path: "/home/cjonrnd/models/research/object_detection/data/pet_label_map.pbtxt"
  shuffle: false
  num_readers: 1
}

------------------------------------------------------------------------------------------------------------------------------
4. I tried running on tensorboard, but gave a 404 error. 
5. Yes, I already tried the tutorial. 

Thanks so much for your help!

**Comment by karmel** [comment]
Tagging @jch1 for more involved help here.

**Comment by nowgood** [comment]
@ashley915 
if you want to use coco evaluation metrics, and  you have copy cocoapi  to models/research directory according to offical suggestion, you could modify `eval_config`  as follow :grinning: 

````
eval_config: {
  num_examples: 2000
  max_evals: 10
  eval_interval_secs: 10
  metrics_set: "coco_detection_metrics"
}
```

**Comment by LXWDL** [comment]
@karmel  Hi，When I use an api to train Pascal VOC datasets, the execution of the eval.py test is stuck there. I use tensorboard to draw a map with only one point instead of a map curve. How do I solve this problem? My model has been trained, the api environment has been configured to test, is OK

**Comment by karmel** [comment]
@LXWDL - I apologize, but I am having a hard time understanding what the problem is, where the problem is, and what version it affects. Please open a new issue for your question and pay attention to the issue template (https://github.com/tensorflow/tensorflow/issues/new). Please provide all the information it asks. Thank you.

@jch1 - thoughts on the original issue here?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
